op-chain-ops/cmd/celo-migrate/state.go

```diff
@@ -208,3 +185,43 @@ func applyStateMigrationChanges(genesis *core.Genesis, dbPath string, commit boo
 
 	return cel2Header, nil
 }
+
+// applyAllocsToState applies the account allocations from the allocation file to the state database.
+// It creates new accounts, sets their nonce, balance, code, and storage values.
+// If an account already exists, it adds the balance of the new account to the existing balance.
+// If the code of an existing account is different from the code in the genesis block, it logs a warning.
+// This changes the state root, so `Commit` needs to be called after this function.
+func applyAllocsToState(db *state.StateDB, genesis *core.Genesis) error {
+	log.Info("Starting to migrate OP contracts into state DB")
+
+	accountCounter := 0
+	overwriteCounter := 0
+	for k, v := range genesis.Alloc {
+		accountCounter++
+
+		balance := uint256.MustFromBig(v.Balance)
+
+		if db.Exist(k) {
+			// If the account already has balance, add it to the balance of the new account
+			balance = balance.Add(balance, db.GetBalance(k))
+
+			equalCode := bytes.Equal(db.GetCode(k), v.Code)
+			if !equalCode {
+				log.Warn("Account already exists with different code, overwriting...", "address", k)
+				overwriteCounter++
+			}
+		}
+		db.CreateAccount(k)
+
+		db.SetNonce(k, v.Nonce)
+		db.SetBalance(k, balance)
+		db.SetCode(k, v.Code)
+		for key, value := range v.Storage {
+			db.SetState(k, key, value)
```

Comment:
Should we have counters / logging for times we overwrite storage or nonce?

Reply:
I need to add a whitelist here as well, will move this out of this PR